lib/sspi/sspwin32.cc

```diff
@@ -9,9 +9,23 @@
 #include "squid.h"
 #include "base64.h"
 #include "ntlmauth/ntlmauth.h"
-#include "sspwin32.h"
+#include "sspi/sspwin32.h"
 #include "util.h"
 
+#if HAVE_WINDOWS_H && HAVE_SSPI_H
+
+// FARPROC is an exception on Windows to the -Wcast-function-type sanity check.
+// suppress the warning only when casting FARPROC
+template<typename T>
```

Comment:
To improve (IMO) readability, I recommend using the following (more prevalent in Squid code) formatting for this line:

```suggestion
template <typename T>
```